### ISSUE TITLE
Error messages like in Reason

### ISSUE THREAD

**Issue by ShalokShalom** [fsharp/FSharp.Compiler.Service#801]
This looks very tasty to me: 
https://reasonml.github.io/community/blog/#way-way-waaaay-nicer-error-messages

It is a new implementation of error messages for Reason and Bucklescript, same like in Elm also.

**Comment by forki** [comment]
There is a big community driven push for better error messages in F# as
well. See https://github.com/Microsoft/visualfsharp/issues/1103 for
details. As you can see things are improving and you can be a part of it by
proposing concrete improvements or sending imp via pull request.  So we're
actually a step further than this generic request for "give us better
errors" ;-)

To address the linked article:


For F# we need a different approach.  The compiler is usually not used from
terminal but from within VS or ionide or other IDE. So we need to optimize
for that. This differs a lot. Many things are actually done by red
squigglies instead of copying code to the console. For most people this is
even much much more useful. Often we now even provide quickfixes that fix
the error for you when you click OK. This works surprisingly well in
practice for things like typos and I would never ever want to go back to
copy fixes from some console output.

In short: we need to address concrete error messages and suggestions for
better wording. And we should take IDE features into account to see that F#
is in many cases already very well equipped.

Am 07.09.2017 10:04 schrieb "ShalokShalom" <notifications@github.com>:

**Comment by ShalokShalom** [comment]
Sounds nice, thanks. :+1:
